vortex-scan/src/range_scan.rs

```diff
@@ -93,4 +96,35 @@ impl RangeScan {
         }
         Ok(())
     }
+
+    /// Evaluate the [`RangeScan`] operation using a synchronous expression evaluator.
+    pub fn evaluate(mut self, evaluator: &dyn Evaluator) -> VortexResult<ArrayData> {
+        loop {
+            match self.next() {
+                NextOp::Ready(array) => return Ok(array),
+                NextOp::Eval((row_range, mask, expr)) => {
+                    self.post(evaluator.evaluate(RowMask::new(mask, row_range.start), expr)?)?;
+                }
+            }
+        }
+    }
+
+    /// Evaluate the [`RangeScan`] operation using an async expression evaluator.
+    pub async fn evaluate_async(
+        mut self,
+        evaluator: &dyn AsyncEvaluator,
+    ) -> VortexResult<ArrayData> {
+        loop {
+            match self.next() {
+                NextOp::Ready(array) => return Ok(array),
+                NextOp::Eval((row_range, mask, expr)) => {
```

Comment:
This code also looks like us manually implementing async, yeah? I see you're also trying to support a sync evaluator so maybe there's not much to be done.

Reply:
Sort of yes, although each operation requires a result to be posted back. We could do this by handing out closures, but 🤷 it's internal to this struct now so I'm less fussed. Plus it's the naive impl, you guys are going to change it a lot I'm sure.